src/Symfony/Component/HttpClient/Response/AmpResponse.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Amp\Http\Client\Request;
+use Amp\Http\Client\Response;
+use Amp\Loop;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpClient\Chunk\FirstChunk;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\Internal\AmpClientState;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+final class AmpResponse implements ResponseInterface
+{
+    use ResponseTrait;
+
+    private $multi;
+    private $request;
+    private $shouldBuffer;
+
+    /**
+     * @internal
+     */
+    public function __construct(AmpClientState $multi, Request $request, array $options, ?LoggerInterface $logger)
+    {
+        $this->multi = $multi;
+        $this->logger = $logger;
+        $this->timeout = $options['timeout'];
+        $this->info = [
+            'response_headers' => [],
+            'url' => (string) $request->getUri(),
```

Comment:
If passwords are allowed in the URI, this will dump the password here.

Reply:
actually passwords are removed from the URL in prepareRequest()

Reply:
After facts checking: dumping passwords here is consistent with the other implementations.